nsls2forge_utils/auto_tick.py

```diff
@@ -0,0 +1,627 @@
+                        hash_type=attrs.get("hash_type", "sha256"),
+                        fork=fork,
+                        organization=organization
+                    )
+                    # if migration successful
+                    if migrator_uid:
+                        d = frozen_to_json_friendly(migrator_uid)
+                        # if we have the PR already do nothing
+                        if d["data"] in [
+                            existing_pr["data"] for existing_pr in attrs.get("PRed", [])
+                        ]:
+                            pass
+                        else:
+                            if not pr_json:
+                                pr_json = {
+                                    "state": "closed",
+                                    "head": {"ref": "<this_is_not_a_branch>"},
+                                }
+                            d["PR"] = pr_json
+                            attrs.setdefault("PRed", []).append(d)
+                        attrs.update(
+                            {
+                                "smithy_version": mctx.smithy_version,
+                                "pinning_version": mctx.pinning_version,
+                            },
+                        )
+
+                except github3.GitHubError as e:
+                    if e.msg == "Repository was archived so is read-only.":
+                        attrs["archived"] = True
+                    else:
+                        logger.critical(
+                            "GITHUB ERROR ON FEEDSTOCK: %s", fctx.feedstock_name,
+                        )
+                        if is_github_api_limit_reached(e, mctx.gh):
+                            break
+                except URLError as e:
+                    logger.exception("URLError ERROR")
+                    attrs["bad"] = {
+                        "exception": str(e),
+                        "traceback": str(traceback.format_exc()).split("\n"),
+                        "code": getattr(e, "code"),
+                        "url": getattr(e, "url"),
+                    }
+                except Exception as e:
+                    logger.exception("NON GITHUB ERROR")
+                    attrs["bad"] = {
+                        "exception": str(e),
+                        "traceback": str(traceback.format_exc()).split("\n"),
+                    }
+                else:
+                    if migrator_uid:
+                        # On successful PR add to our counter
+                        good_prs += 1
+                finally:
+                    # Write graph partially through
+                    if not dry_run:
+                        dump_graph(mctx.graph)
+
+                    eval_cmd(f"rm -rf {mctx.rever_dir}/*")
+                    logger.info(os.getcwd())
+                    for f in glob.glob("/tmp/*"):
+                        if f not in temp:
+                            eval_cmd(f"rm -rf {f}")
```

Comment:
I don't fully understand the logic here. Why do we keep this file if it in `/tmp`?

Reply:
I am not sure either, I figured they use it as some sort of save for certain files they don't want affected by the process. I haven't seen it used anywhere else though and it doesn't show up when I run the bot anyway.